components/balance/MultipleBalances.vue

```diff
@@ -0,0 +1,172 @@
+import { decodeAddress, encodeAddress } from '@polkadot/util-crypto'
+import { CHAINS, ENDPOINT_MAP } from '@kodadot1/static'
+import { NeoSkeleton } from '@kodadot1/brick'
+import { balanceOf } from '@kodadot1/sub-api'
+
+import format from '@/utils/format/balance'
+import { useFiatStore } from '@/stores/fiat'
+
+import type { PalletBalancesAccountData } from '@polkadot/types/lookup'
+import { calculateExactUsdFromToken } from '@/utils/calculation'
+
+interface Token {
+  token: string
+  balance: string
+  usd: number
+}
+
+interface Chain {
+  [key: string]: Token[]
+}
+
+const { accountId } = useAuth()
+
+const status = ref('loading')
+const availableChains = reactive<Chain>({
+  Kusama: [],
+  Statemine: [],
+  Basilisk: [],
+})
+
+const mapToPrefix = {
+  Kusama: 'ksm',
+  Basilisk: 'bsx',
+  Statemine: 'stmn',
+}
+
+function delimiter(amount: string | number) {
+  const formatAmount = typeof amount === 'number' ? amount.toString() : amount
+  const number = parseFloat(formatAmount.replace(/,/g, ''))
+
+  return number.toLocaleString('en-US', {
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  })
+}
+
+const fiatStore = useFiatStore()
+const totalUsd = ref([0])
+const sumUsd = computed(() => totalUsd.value.reduce((a, b) => a + b, 0))
+function calculateUsd(amount: string, token = 'KSM') {
+  if (!amount) {
+    return 0
+  }
+
+  const amountToNumber = Number(amount.replace(/\,/g, ''))
+
+  if (token === 'KSM') {
+    return calculateExactUsdFromToken(
+      amountToNumber,
+      Number(fiatStore.getCurrentKSMValue)
+    )
+  }
+
+  return calculateExactUsdFromToken(
+    amountToNumber,
+    Number(fiatStore.getCurrentBSXValue)
+  )
+}
+
+async function getBalance(chainName, token = 'KSM', tokenId = 0) {
+  const currentAddress = accountId.value
+  const prefix = mapToPrefix[chainName]
+  const chain = CHAINS[prefix]
+
+  const publicKey = decodeAddress(currentAddress)
+  const prefixAddress = encodeAddress(publicKey, chain.ss58Format)
+  const wsProvider = new WsProvider(ENDPOINT_MAP[prefix])
+
+  const api = await ApiPromise.create({
+    provider: wsProvider,
+  })
+
+  let currentBalance
+
+  if (tokenId) {
+    const balance = await api.query.tokens.accounts(prefixAddress, tokenId)
+    currentBalance = format(
+      (balance as PalletBalancesAccountData).free,
+      chain.tokenDecimals,
+      false
+    )
+  } else {
+    const balance = await balanceOf(api, prefixAddress)
+    currentBalance = format(balance, chain.tokenDecimals, false)
+  }
+
+  const balance = delimiter(currentBalance)
+  const usd = calculateUsd(balance, token)
+
+  availableChains[chainName].push({
+    token,
+    balance,
+    usd,
+  })
+  totalUsd.value.push(usd)
+
+  await wsProvider.disconnect()
+}
+
+onMounted(async () => {
+  await fiatStore.fetchFiatPrice()
+
+  Promise.all([
+    getBalance('Kusama'),
+    getBalance('Statemine'),
+    getBalance('Basilisk', 'BSX'),
+    getBalance('Basilisk', 'KSM', 1),
```

Comment:
1 is for snek?

would be nice to have somewhere a map of token names to their id's instead of magic numbers

Reply:
maybe https://github.com/kodadot/nft-gallery/blob/0ea86c3f1d3e97bb7827b577077c6bf5b6005fb3/utils/api/bsx/query.ts#L69 ?